spec/mocks/console_mock.rb

```diff
@@ -0,0 +1,52 @@
+class ConsoleMock
```

Comment:
This class is now big enough with enough logic, to warrant its own test suite. This is probably a Fake. Which is actually ok to test, but in this case, I think we can remove this and use StringIO to help solve the problems we have.